php/boot-phar.php

```diff
@@ -1,5 +1,8 @@
 <?php
 
+// Store the path to the Phar early on for `Utils\phar-safe-path()` function.
+define( 'WP_CLI_PHAR_PATH', getcwd() );
```

Comment:
The mechanism works by having something to diff against `__DIR__`.

```
__DIR__ => phar:///home/alain/test/wp-cli.phar/php
getcwd() => /home/alain/test
```

This gives us the bit in front of the phar name, and it even works with different phar names.